// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.scp.cpio.validator.proto;

message ValidatorConfig {
  repeated TestCase test_cases = 1;
}

message TestCase {
  // User provided (preferrably unique) name for the test case.
  string name = 1;

  oneof client_config {
    GetTagsByResourceNameConfig get_tags_by_resource_name_config = 2;
    GetCurrentInstanceResourceNameConfig get_current_instance_resource_name_config = 3;
    GetBlobConfig get_blob_config = 4;
    ListBlobsMetadataConfig list_blobs_metadata_config = 5;
    GetParameterConfig get_parameter_config = 6;
    EnqueueMessageConfig enqueue_message_config = 7;
    GetTopMessageConfig get_top_message_config = 8;
  }
}

message GetTagsByResourceNameConfig {
  string resource_name = 1;
}

message GetCurrentInstanceResourceNameConfig {}

message GetBlobConfig {
  string bucket_name = 1;
  string blob_name = 2;
}

message ListBlobsMetadataConfig {
  string bucket_name = 1;
}

message GetParameterConfig {
  string parameter_name = 1;
}

message EnqueueMessageConfig {
  string message_body = 1;
}

message GetTopMessageConfig {}
